lua/lsp/init.lua

```diff
@@ -128,22 +128,19 @@ function M.common_on_attach(client, bufnr)
 end
 
 function M.setup(lang)
-  local lsp_utils = require "lsp.utils"
-  local lsp = lvim.lang[lang].lsp
-  if lsp_utils.is_client_active(lsp.provider) then
-    return
-  end
-
   local overrides = lvim.lsp.override
   if type(overrides) == "table" then
     if vim.tbl_contains(overrides, lang) then
       return
     end
   end
 
-  if lsp.provider ~= nil and lsp.provider ~= "" then
-    local lspconfig = require "lspconfig"
-    lspconfig[lsp.provider].setup(lsp.setup)
+  -- initialize language server for each provider
+  local providers = lvim.lang[lang].lsp.providers
+  for _, provider in pairs(providers) do
+    if provider.name ~= nil and provider.name ~= "" and not require("lsp.utils").is_client_active(provider.name) then
```

Comment:
I know you just adapted the existing code, but our code didn't make sens
can you do
``` lua
local lsp_utils = require("lsp.utils") -- Put above providers
if not lsp_utils.is_client_active(provider.name) then
   ....
end
```

Can you then do a pass on default-config to change:

``` lua
      providers = {
        {
          name = "",
          setup = {},
        },
      },
```

into

``` lua
 providers = {},
```

Reply:
changed

Reply:
Please resolve the conversation :)